src/azure-cli/azure/cli/command_modules/resource/_params.py

```diff
@@ -190,15 +190,17 @@ def load_arguments(self, _):
 
     with self.argument_context('group deployment create') as c:
         c.argument('deployment_name', arg_type=deployment_create_name_type)
-        c.argument('handle_extended_json_format', arg_type=extended_json_format_type)
+        c.argument('handle_extended_json_format', arg_type=extended_json_format_type,
```

Comment:
you can only add deprecate_info in extended_json_format_type.

Reply:
Yes, that's what I wanted to write at the beginning.  But I found that this would cause an error if the specific command's argument_context was not bound. 
If you have a good way to solve this problem, please share with me, thank you~